gradio/blocks.py

```diff
@@ -2482,29 +2482,51 @@ def get_api_info(self):
                 continue
 
             dependency_info = {"parameters": [], "returns": []}
+            fn_info = utils.get_function_params(fn.fn)  # type: ignore
             skip_endpoint = False
 
             inputs = dependency["inputs"]
-            for i in inputs:
+            for index, input_id in enumerate(inputs):
                 for component in config["components"]:
-                    if component["id"] == i:
+                    if component["id"] == input_id:
                         break
                 else:
                     skip_endpoint = True  # if component not found, skip endpoint
                     break
                 type = component["props"]["name"]
                 if self.blocks[component["id"]].skip_api:
                     continue
-                label = component["props"].get("label", f"parameter_{i}")
+                label = component["props"].get("label", f"parameter_{input_id}")
                 comp = self.get_component(component["id"])
                 if not isinstance(comp, components.Component):
                     raise TypeError(f"{comp!r} is not a Component")
                 info = component["api_info"]
                 example = comp.example_inputs()
                 python_type = client_utils.json_schema_to_python_type(info)
+
+                # Since the clients use "api_name" and "fn_index" to designate the endpoint and
+                # "result_callbacks" to specify the callbacks, we need to make sure that no parameters
+                # have those names. Hence the final checks.
+                if (
+                    dependency["backend_fn"]
+                    and index < len(fn_info)
+                    and fn_info[index][0]
+                    not in ["api_name", "fn_index", "result_callbacks"]
+                ):
+                    parameter_name = fn_info[index][0]
+                    parameter_has_default = fn_info[index][1]
+                    parameter_default = fn_info[index][2]
```

Comment:
The default value may not be json-serializable. We should check against that return `None` in that case.